src/server/api/url_parser.cpp

```diff
@@ -1,88 +1,78 @@
 #include "server/api/url_parser.hpp"
-
 #include "engine/polyline_compressor.hpp"
 
-#include <boost/bind.hpp>
-#include <boost/spirit/include/qi_char_.hpp>
-#include <boost/spirit/include/qi_grammar.hpp>
-#include <boost/spirit/include/qi_uint.hpp>
-#include <boost/spirit/include/qi_real.hpp>
-#include <boost/spirit/include/qi_lit.hpp>
-#include <boost/spirit/include/qi_action.hpp>
-#include <boost/spirit/include/qi_as_string.hpp>
-#include <boost/spirit/include/qi_operator.hpp>
-#include <boost/spirit/include/qi_plus.hpp>
```

Comment:
In theory yes. In practice, the grammar is only used in this single cpp file and therefore is TU-local.

For grammars in headers what helps is explicitely instantiating the grammar as in http://stackoverflow.com/a/5507332.